lib/_http_server.js

```diff
@@ -560,7 +570,7 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
     }
   }
 
-  var res = new ServerResponse(req);
+  var res = new socket.server.ServerResponse(req);
```

Comment:
Could this just be `server.serverResponse(req)` instead of `socket.server[...]`? The function seems to have an in-scope `server`.

Reply:
I tried but we don't have the `server` here, only under the `socket.server`: https://github.com/nodejs/node/blob/42a2a9badb6974e47b98aaee052e9710f34b93e8/lib/_http_common.js#L77

Reply:
I could swear it's available 

```js
function parserOnIncoming(server, socket, state, req, keepAlive) {
  resetSocketTimeout(server, socket, state);
```

It's being used a few places in that function to `emit`, etc.

Reply:
@hekike it's in the function definition, it should be populated. Can you update?

Reply:
@mcollina @apapirovski thanks, fixed!